cpp-ch/CMakeLists.txt

```diff
@@ -27,93 +29,93 @@ message("CH_COMMIT=${CH_COMMIT}")
+# execute_process(COMMAND find ${CMAKE_CURRENT_SOURCE_DIR}/local-engine -regex
+# '.*\.\(c\|cpp\|h\)' -exec clang-format-15 --verbose -i --style=file -i {} \;)
 
 set(CH_BINARY_DIR "${CMAKE_CURRENT_SOURCE_DIR}/build")
 option(ENABLE_CPP_TEST "Build CPP Unit test" OFF)
 
-if (ENABLE_CPP_TEST)
-add_custom_command(
-        USES_TERMINAL
-        COMMAND
-        bash -c
-        \"cmake -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
-        -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
-        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
-        -DENABLE_PROTOBUF=ON
-        -DENABLE_TESTS=ON
-        -DENABLE_JEMALLOC=ON
-        -DENABLE_MULTITARGET_CODE=ON
-        -DENABLE_EXTERN_LOCAL_ENGINE=ON
-        -DCOMPILER_FLAGS='-fvisibility=hidden -fvisibility-inlines-hidden'
-        -S ${CH_SOURCE_DIR} -G Ninja -B ${CH_BINARY_DIR} &&
-        cmake --build ${CH_BINARY_DIR} --target ch unit_tests_local_engine\"
-        OUTPUT _build_ch)
+if(ENABLE_CPP_TEST)
+  add_custom_command(
+    USES_TERMINAL
+    COMMAND
+      bash -c \"cmake -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+      -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+      -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} -DENABLE_PROTOBUF=ON
+      -DENABLE_TESTS=ON -DENABLE_JEMALLOC=ON -DENABLE_MULTITARGET_CODE=ON
+      -DENABLE_EXTERN_LOCAL_ENGINE=ON -DCOMPILER_FLAGS='-fvisibility=hidden
+      -fvisibility-inlines-hidden' -S ${CH_SOURCE_DIR} -G Ninja -B
+      ${CH_BINARY_DIR} && cmake --build ${CH_BINARY_DIR} --target ch
+      unit_tests_local_engine\"
+    OUTPUT _build_ch)
 else()
-add_custom_command(
-        USES_TERMINAL
-        COMMAND
-        bash -c
-        \"cmake -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
-        -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
-        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
-        -DENABLE_PROTOBUF=ON
-        -DENABLE_TESTS=OFF
-        -DENABLE_JEMALLOC=ON
-        -DENABLE_MULTITARGET_CODE=ON
-        -DENABLE_EXTERN_LOCAL_ENGINE=ON
-        -DENABLE_ODBC=OFF
-        -DENABLE_CAPNP=OFF
-        -DENABLE_ROCKSDB=OFF
-        -DENABLE_GRPC=OFF
-        -DENABLE_RUST=OFF
-        -DENABLE_H3=OFF
-        -DENABLE_AMQPCPP=OFF
-        -DENABLE_CASSANDRA=OFF
-        -DENABLE_KAFKA=OFF
-        -DENABLE_NATS=OFF
-        -DENABLE_LIBPQXX=OFF
-        -DENABLE_NURAFT=OFF
-        -DENABLE_DATASKETCHES=OFF
-        -DENABLE_SQLITE=OFF
-        -DENABLE_S2_GEOMETRY=OFF
-        -DENABLE_ANNOY=OFF
-        -DENABLE_ULID=OFF
-        -DENABLE_MYSQL=OFF
-        -DENABLE_BCRYPT=OFF
-        -DENABLE_LDAP=OFF
-        -DENABLE_MSGPACK=OFF
-        -DUSE_REPLXX=OFF
-        -DENABLE_CLICKHOUSE_ALL=OFF
-        -DCOMPILER_FLAGS='-fvisibility=hidden -fvisibility-inlines-hidden'
-        -S ${CH_SOURCE_DIR} -G Ninja -B ${CH_BINARY_DIR} &&
-        cmake --build ${CH_BINARY_DIR} --target libch\"
-        OUTPUT _build_ch)
+  add_custom_command(
```

Comment:
It seems to be getting messier here.

Reply:
@liuneng1994, I think for most code, the formatting makes code look better. Maybe, some configuration can be applied in the future to tune the format.